public/js/submitAssess.js

```diff
@@ -336,64 +324,27 @@ $('#assess-button').click(function(event) {
      * 
      ****************************************************************/
 
-    console.log(questionsList);
-    console.log(doBetterQuestions);
-    
-    // Existing market.
-    if (responses.length == 4) {
-        console.log("push existing market");
-        // Check if new market or existing market.
-        sessionStorage.setItem("formname",responses[3].value);
-        sessionStorage.setItem("lvl", marketLevel.toString());
-
-        var marketName = responses[3].value;
-
-        marketsRef = marketsRef.child(marketName);
-
-        // Update market level.
-        marketsRef.child("marketInfo").update({
-            marketLevel: marketLevel
-        });
-        // Update user info.
-        marketsRef.child("personalInfo").update({
-            firstName: responses[0].value,
-            lastName: responses[1].value,
-            email: responses[2].value
-        });
-        // Update question responses.
-        marketsRef.child("questions").set(questionsList);
-        marketsRef.child("missedQuestions").set(doBetterQuestions);
-
-    } else { // New market.
-        console.log("push new market");
-        // Check if new market or existing market.
-        sessionStorage.setItem("formname",responses[4].value);
-        sessionStorage.setItem("lvl", marketLevel.toString());
-
-        var marketName = responses[4].value + ', ' + responses[6].value;
-        // Make sure illegal characters removed from key.
-        marketName = marketName.replace(/[^0-9a-zA-Z, ]/gi, '')
-
-        marketsRef.child(marketName).set({
-            personalInfo: {
-                firstName: responses[0].value,
-                lastName: responses[1].value,
-                email: responses[2].value,
-            },
-            marketInfo: {
-                marketName: responses[4].value,
-                storeType: responses[5].value,
-                address: responses[6].value,
-                city: responses[7].value,
-                state: responses[8].value,
-                zip: responses[9].value,
-                marketLevel: marketLevel
-            },
-            questions: questionsList,
-            missedQuestions: doBetterQuestions
-        });
+    let sendResponses = {}
+    for (const key in responses) {
+        sendResponses[responses[key].name] = responses[key].value;
     }
 
-    console.log("switching window");
-    location.href='results';
+    let marketExists = "false";
+
+    if (responses.length === 4) {
+        marketExists = "true";
+    }
+
+    const sendData = {
+        existing: marketExists,
+        level: marketLevel,
+        betterQuestions: doBetterQuestions, 
+        marketInfo: sendResponses, 
+        questions: questionsList
+    }
+
+    $.post('/submit-assess', {data: JSON.stringify(sendData)}, function(callback) {});
```

Comment:
1. why does `JSON.stringify` need to be called on `sendData`?
2. if the callback doesn't do anything, can it be safely deleted?

Reply:
Stringify has to be called otherwise the JSON gets all kinds of screwed up on the other end.  That has nested JSON in it which for some reason screwed up all the keys, which is why I rebuilt it above.

I'll get rid of the empty callback